frontend/src/lib/k8s/apiProxy.ts

```diff
@@ -45,8 +49,24 @@ export async function request(path: string, params: RequestParams = {},
     fullPath = combinePath(`/${CLUSTERS_PREFIX}/${cluster}`, path);
   }
 
+  const controller = new AbortController();
+  const id = setTimeout(() => controller.abort(), timeout);
+
   const url = combinePath(BASE_HTTP_URL, fullPath);
-  const response = await fetch(url, opts);
+  let response: Response = new Response(undefined,
+                                        {status: 502, statusText: 'Unreachable'});
+
+  try {
+    response = await fetch(url, {signal: controller.signal, ...opts});
+  } catch(err) {
+    // AbortController sets the error code as 20
```

Comment:
```suggestion
    // AbortController sets the error code as 20.
```

Reply:
I use periods for comments that are longer than one line. One single line is optional.